src/rust/tests/test_index.rs

```diff
@@ -0,0 +1,476 @@
+use anyhow::Result;
+use sourmash::encodings::HashFunctions;
+use sourmash::signature::SigsTrait;
+use sourmash::sketch::minhash::KmerMinHash;
+use sourmash_plugin_branchwater::utils::multicollection::SmallSignature;
+
+use tempfile::tempdir;
+
+use crate::index::ProteomeIndex;
+use crate::tests::test_fixtures::{TEST_KMER, TEST_PROTEIN};
+use std::collections::HashMap;
+
+#[test]
+fn test_kmer_encoding() -> Result<()> {
+    let dir = tempdir()?;
+
+    // Create index with minimal parameters
+    let index = ProteomeIndex::new(
+        dir.path().join("kmer_test.db"),
+        11, // small ksize for test
+        1,  // scaled
+        "protein",
+    )?;
+
+    // Test k-mer encoding
+    let (encoded_kmer, original_kmer) = index.encode_kmer(TEST_KMER);
+
+    // The original k-mer should be exactly what we put in
+    assert_eq!(original_kmer, TEST_KMER);
+
+    // The encoded k-mer should be the same as the original k-mer
+    assert_eq!(encoded_kmer, TEST_KMER);
+
+    Ok(())
+}
+
+#[test]
+fn test_kmer_encoding_dayhoff() -> Result<()> {
+    let dir = tempdir()?;
+
+    // Create index with minimal parameters
+    let index = ProteomeIndex::new(
+        dir.path().join("kmer_test.db"),
+        11, // small ksize for test
+        1,  // scaled
+        "dayhoff",
+    )?;
+
+    // Test k-mer encoding
+    let (encoded_kmer, original_kmer) = index.encode_kmer(TEST_KMER);
+
+    // The original k-mer should be exactly what we put in
+    assert_eq!(original_kmer, TEST_KMER);
+
+    // The encoded k-mer should be dayhoff-encoded k-mer
+    assert_eq!(encoded_kmer, "eeeecbbeeec");
+
+    Ok(())
+}
+
+#[test]
+fn test_kmer_encoding_hp() -> Result<()> {
+    let dir = tempdir()?;
+
+    // Create index with minimal parameters
+    let index = ProteomeIndex::new(
+        dir.path().join("kmer_test.db"),
+        11, // small ksize for test
+        1,  // scaled
+        "hp",
+    )?;
+
+    // Test k-mer encoding
+    let (encoded_kmer, original_kmer) = index.encode_kmer(TEST_KMER);
```

Comment:
```suggestion
    let (encoded_kmer, original_kmer) = index.encode_kmer(TEST_KMER)?;
```
only if the encode_kmer() suggestion is accepted